### PR TITLE
 Fixed issues with radix sort implementation

### LINKED ISSUE #20: "AlgorithmsRadixSort" can't sort long properly, and also fails under CPU
It will give something like following result (negative is listed behind positive) if there are negative long in the array.
1,2,3,4,5,-5,-4,-3,-2,-1

It also throws error message if accelarator is CPU

### PR DESCRIPTION
**Pull request by MoFtZ**
Fixes https://github.com/m4rs-mt/ILGPU.Algorithms/issues/20

### CODE REVIEW

File: Src/ILGPU.Algorithms/RadixSortExtensions.cs [hunk 1]
```diff
@@ -132,6 +132,7 @@ public static class RadixSortExtensions
             KernelConfig config,
             ArrayView<T> view,
             ArrayView<int> counter,
+            SpecializedValue<int> groupSize,
```
Comment (m4rs-mt) [inline Src/ILGPU.Algorithms/RadixSortExtensions.cs:135]: A perfect use case for the latest compiler features 🚀

**Review by m4rs-mt — APPROVED** [review]
An awesome piece of work 🥇 Thanks a lot for submitting this patch.

**Comment by GeoBIM2020** [comment]
@MoFtZ Can RadixSort deal with signed long and double?

**Comment by MoFtZ** [comment]
hi @toolwtech, yes, this PR includes support for signed integers (sbyte, short, int, long) and floating point numbers (float, double).

**Pull request merged**